projects/gloo/cli/pkg/cmd/gatewayapi/convert/command.go

```diff
@@ -0,0 +1,223 @@
+package convert
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/solo-io/gloo/pkg/utils/envoyutils/admincli"
+	"github.com/solo-io/gloo/pkg/utils/kubeutils/kubectl"
+	"github.com/solo-io/gloo/pkg/utils/kubeutils/portforward"
+	"github.com/solo-io/gloo/pkg/utils/requestutils/curl"
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/cmd/options"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	RandomSuffix = 4
+	RandomSeed   = 1
+)
+
+func RootCmd(op *options.Options) *cobra.Command {
+	opts := &Options{
+		Options: op,
+	}
+	cmd := &cobra.Command{
+		Use:   "convert",
+		Short: "Convert Gloo Edge APIs to Gateway API",
+		Long:  "Convert Gloo Edge APIs to Gateway API by either providing kubernetes yaml files or a Gloo Gateway input snapshot",
+		Example: `# This command converts Gloo Edge APIs to Gloo Gateway API yaml and places them in the '--output-dir' directory arranged by namespace.
+# To generate gateway api by getting snapshot directly from running Gloo pod. The 'output-dir'' must not exist
+  glooctl gateway-api convert --gloo-control-plane deploy/gloo --output-dir ./_output
+
+# To generate gateway api by a single kubernetes yaml file. The 'output-dir'' must not exist
+  glooctl gateway-api convert --input-file gloo-yamls.yaml --output-dir ./_output
+
+# To delete and recreate the output dir add 'delete-output-dir'
+  glooctl gateway-api convert --input-file gloo-yamls.yaml --output-dir ./_output --delete-output-dir
+
+# To generate gateway api by a single kubernetes yaml file but place all the output configurations in the same file.
```

Comment:
does this mean, I will end up with one gigantic file? if so, the `retain-input-foder-structure` is a bit confusing as I am using a single file as an input

Reply:
thats wierd for snapshots yes but its mainly for users that run this tool against CI/CD git repos and want to keep the folder structure